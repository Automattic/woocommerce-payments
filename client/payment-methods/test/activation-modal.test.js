/** @format */

/**
 * External dependencies
 */
import React from 'react';
import { render, screen } from '@testing-library/react';
import user from '@testing-library/user-event';

/**
 * Internal dependencies
 */
import ConfirmPaymentMethodActivationModal from '../activation-modal';

const mockOnClose = jest.fn();
const mockOnConfirm = jest.fn();

describe( 'Activation Modal', () => {
	beforeAll( () => {
		wcSettings = {
			currentUserData: {
				email: 'foo@bar.com',
			},
		};
	} );
	afterEach( () => {
		jest.resetAllMocks();
	} );

	const renderActivationModal = ( requirements ) => {
		return render(
			<ConfirmPaymentMethodActivationModal
				paymentMethod={ 'card' }
				onClose={ mockOnClose }
				onConfirmClose={ mockOnConfirm }
				requirements={ requirements }
			/>
		);
	};

	it( 'matches the snapshot', () => {
		const container = renderActivationModal( [ 'company.tax_id' ] )
			.baseElement;
		expect( container ).toMatchSnapshot();
	} );

	it( 'contains the requirement text', () => {
		renderActivationModal( [ 'company.tax_id' ] );
		expect( screen.queryByText( 'Business Number' ) ).toBeInTheDocument();
	} );

	it( 'contains the email address of the current merchant', () => {
		renderActivationModal( [ 'company.tax_id' ] );
		expect( screen.queryByText( 'foo@bar.com' ) ).toBeInTheDocument();
	} );

	it( 'doesnt contain the requirement text when there isnt a translation', () => {
		renderActivationModal( [ 'person.tax_id' ] );
		expect( screen.queryByText( 'person.tax_id' ) ).not.toBeInTheDocument();
	} );

	it( 'triggers the onClose event on close button click', () => {
		renderActivationModal( [] );
		const closeButton = screen.queryByRole( 'button', { name: 'Cancel' } );
		expect( mockOnClose ).not.toBeCalled();
		user.click( closeButton );
		expect( mockOnClose ).toBeCalled();
	} );

	it( 'triggers the onConfirmClose event on confirm button click', () => {
		renderActivationModal( [] );
		const confirmButton = screen.queryByRole( 'button', {
			name: 'Continue',
		} );
		expect( mockOnConfirm ).not.toBeCalled();
		user.click( confirmButton );
		expect( mockOnConfirm ).toBeCalled();
	} );
} );
