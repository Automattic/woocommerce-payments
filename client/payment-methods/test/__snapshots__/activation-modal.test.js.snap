// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Activation Modal matches the snapshot 1`] = `
<body
  class="modal-open"
>
  <p
    aria-hidden="true"
    class="a11y-speak-intro-text"
    hidden="hidden"
    id="a11y-speak-intro-text"
    style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
  >
    Notifications
  </p>
  <div
    aria-atomic="true"
    aria-live="assertive"
    aria-relevant="additions text"
    class="a11y-speak-region"
    id="a11y-speak-assertive"
    style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
  />
  <div
    aria-atomic="true"
    aria-live="polite"
    aria-relevant="additions text"
    class="a11y-speak-region"
    id="a11y-speak-polite"
    style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
  />
  <div
    aria-hidden="true"
  />
  <div>
    <div>
      <div>
        <div
          tabindex="-1"
        >
          <div>
            <div
              class="components-modal__screen-overlay"
            >
              <div
                aria-labelledby="components-modal-header-0"
                class="components-modal__frame wcpay-confirmation-modal wcpay-payment-method-confirmation-modal"
                role="dialog"
                tabindex="-1"
              >
                <div
                  class="components-modal__content"
                  role="document"
                >
                  <div
                    class="components-modal__header"
                  >
                    <div
                      class="components-modal__header-heading-container"
                    >
                      <h1
                        class="components-modal__header-heading"
                        id="components-modal-header-0"
                      >
                        One more step to enable Credit card / debit card
                      </h1>
                    </div>
                    <button
                      aria-label="Close dialog"
                      class="components-button has-icon"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        focusable="false"
                        height="24"
                        role="img"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M13 11.8l6.1-6.3-1-1-6.1 6.2-6.1-6.2-1 1 6.1 6.3-6.5 6.7 1 1 6.5-6.6 6.5 6.6 1-1z"
                        />
                      </svg>
                    </button>
                  </div>
                  <div
                    class="payment-confirm-illustration__wrapper"
                  >
                    <div
                      class="payment-confirm-illustration__illustrations"
                    >
                      <img
                        alt="Credit Card"
                        class="payment-confirm-illustration__payment-icon"
                        src="assets/images/payment-methods/cc.svg"
                      />
                      <svg
                        class="gridicon gridicons-help payment-confirm-illustration__payment-question-mark-icon"
                        height="24"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <g>
                          <path
                            d="M12 2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10S17.523 2 12 2zm1 16h-2v-2h2v2zm0-4.14V15h-2v-2c0-.552.448-1 1-1 1.103 0 2-.897 2-2s-.897-2-2-2-2 .897-2 2H8c0-2.21 1.79-4 4-4s4 1.79 4 4c0 1.862-1.278 3.413-3 3.86z"
                          />
                        </g>
                      </svg>
                    </div>
                  </div>
                  <p>
                    You need to provide more information to enable Credit card / debit card on your checkout:
                  </p>
                  <ul
                    class="payment-method-requirements-list"
                  >
                    <li>
                      Business Number
                    </li>
                  </ul>
                  <p>
                    If you choose to continue, our payment partner Stripe will send an e-mail to 
                    <b>
                      admin@example.com
                    </b>
                     to collect the required information
                  </p>
                  <hr
                    class="wcpay-confirmation-modal__separator"
                  />
                  <div
                    class="wcpay-confirmation-modal__footer"
                  >
                    <button
                      class="components-button is-secondary"
                      type="button"
                    >
                      Cancel
                    </button>
                    <button
                      class="components-button is-primary"
                      type="button"
                    >
                      Continue
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
